lib/index.ts

```diff
@@ -0,0 +1,244 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE
+
+const lighthouse = require('lighthouse');
+const ChromeLauncher = require('lighthouse/lighthouse-cli/chrome-launcher');
+const perfConfig: any = require('lighthouse/lighthouse-core/config/perf.json');
+const opn = require('opn');
+
+const Sheets = require('./sheets/index');
+const Chart = require('./chart/chart');
+const metrics = require('./metrics');
+const expectations = require('./expectations');
+const {upload} = require('./upload');
+const messages = require('./utils/messages');
+
+import {MainOptions, FeatureFlags, AuthorizeCredentials, LighthouseResults, MetricsResults, DriveResponse, TermWritableStream} from '../types/types';
+
+const _MAX_LIGHTHOUSE_TRIES = 2;
+const _SIGINT = 'SIGINT';
+const _SIGINT_EXIT_CODE = 130;
+
+class PWMetrics {
+  flags: FeatureFlags|any;
+  runs: number;
+  sheets: Object;
+  expectations: Object;
+  clientSecret: AuthorizeCredentials;
+  tryLighthouseCounter: number;
+  launcher: any;
+
+  constructor(public url: string, opts: MainOptions) {
+    this.flags = opts.flags || {};
+    // @todo remove when new lighthouse version be released, because -https://github.com/GoogleChrome/lighthouse/pull/1778
+    this.flags.disableCpuThrottling = this.flags.disableCpuThrottling || false;
+    this.runs = this.flags.runs || 1;
+    this.sheets = opts.sheets;
+    this.expectations = opts.expectations || false;
+    this.clientSecret = opts.clientSecret;
+    this.tryLighthouseCounter = 0;
+
+    if (this.flags.expectations) {
+      if (this.expectations) {
+        expectations.validateMetrics(this.expectations);
+        this.expectations = expectations.normalizeMetrics(this.expectations);
+      } else throw new Error(getMessageWithPrefix('ERROR', 'NO_EXPECTATIONS_FOUND'));
+    }
+  }
+
+  async start() {
+    const runs = Array.apply(null, {length: this.runs}).map(Number.call, Number);
+    let results: MetricsResults[] = [];
+
+    // Kill spawned Chrome process in case of ctrl-C.
+    process.on(_SIGINT, async () => {
+      if (this.launcher) {
+        await this.launcher.kill();
+        process.exit(_SIGINT_EXIT_CODE);
+        console.log(messages.getMessage('CLOSING_CHROME'));
+      }
+    });
+
+    for (let runIndex of runs) {
+      try {
+        results[runIndex] = await this.run();
+        console.log(messages.getMessageWithPrefix('SUCCESS', 'SUCCESS_RUN', runIndex, runs.length));
+      } catch(error) {
+        results[runIndex] = error;
+        console.error(messages.getMessageWithPrefix('ERROR', 'FAILED_RUN', runIndex, runs.length, error.message));
+      }
+    }
+
+    const ret = {runs: results.filter(r => !(r instanceof Error))};
+    if (this.runs > 1 && !this.flags.submit) {
+      const median = this.findMedianRun(results);
```

Comment:
Change to 
```js
ret.median = this.findMedianRun(ret.runs)
....
this.displayOutput(ret.median);
```
If there is an error the findMedianRun function will not catch it and will crash the program.